File: frontend/src/types/userSafety.ts
export type UserReportReason =
  | "HARASSMENT"
  | "SPAM"
  | "IMPERSONATION"
  | "UNSAFE_BEHAVIOUR"
  | "INAPPROPRIATE_CONTENT"
  | "OTHER";

export interface BlockedUser {
  id: string;
  first_name: string;
  last_name: string;
  avatar_url: string | null;
  blocked_at: string;
}

export interface UserReportPayload {
  reason: UserReportReason;
  details?: string;
}

File: backend/app/schemas/user_safety.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.storage_media import StorageBackedAvatarResponse


ReportReason = Literal[
    "HARASSMENT",
    "SPAM",
    "IMPERSONATION",
    "UNSAFE_BEHAVIOUR",
    "INAPPROPRIATE_CONTENT",
    "OTHER",
]


class UserReportCreate(BaseModel):
    reason: ReportReason
    details: str | None = Field(default=None, max_length=500)


class UserReportResponse(BaseModel):
    id: int
    created_at: datetime
    submitted: bool = True


class UserBlockResponse(BaseModel):
    blocked: bool


class BlockedUserResponse(StorageBackedAvatarResponse):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    first_name: str
    last_name: str
    blocked_at: datetime
